plugins/catalog-backend-module-github/src/providers/GithubOrgEntityProvider.ts

```diff
@@ -224,6 +261,329 @@ export class GithubOrgEntityProvider implements EntityProvider {
+
+    if (!this.canHandleEvents()) {
+      logger.debug(`Skipping event ${params.topic}`);
+      return;
+    }
+
+    const addEntitiesOperation = createAddEntitiesOperation(
+      this.options.id,
+      this.options.gitHubConfig.host,
+    );
+    const removeEntitiesOperation = createRemoveEntitiesOperation(
+      this.options.id,
+      this.options.gitHubConfig.host,
+    );
+    const replaceEntitiesOperation = createReplaceEntitiesOperation(
+      this.options.id,
+      this.options.gitHubConfig.host,
+    );
+
+    // handle change users in the org
+    // https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#organization
+    if (params.topic.includes('organization')) {
+      const orgEvent = params.eventPayload as OrganizationEvent;
+
+      if (
+        orgEvent.action === 'member_added' ||
+        orgEvent.action === 'member_removed'
+      ) {
+        const createDeltaOperation =
+          orgEvent.action === 'member_added'
+            ? addEntitiesOperation
+            : removeEntitiesOperation;
+        await this.onMemberChangeInOrganization(orgEvent, createDeltaOperation);
+      }
+    }
+
+    // handle change teams in the org
+    // https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#team
+    if (params.topic.includes('team')) {
+      const teamEvent = params.eventPayload as TeamEvent;
+      if (teamEvent.action === 'created' || teamEvent.action === 'deleted') {
+        const createDeltaOperation =
+          teamEvent.action === 'created'
+            ? addEntitiesOperation
+            : removeEntitiesOperation;
+        await this.onTeamChangeInOrganization(teamEvent, createDeltaOperation);
+      } else if (teamEvent.action === 'edited') {
+        await this.onTeamEditedInOrganization(
+          teamEvent,
+          replaceEntitiesOperation,
+        );
+      }
+    }
+
+    // handle change membership in the org
+    // https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#membership
+    if (params.topic.includes('membership')) {
+      const membershipEvent = params.eventPayload as MembershipEvent;
+      const updateOperation =
+        membershipEvent.action === 'added'
+          ? addUserToGroup
+          : removeUserFromGroup;
+      await this.onMemberChangeToTeam(
+        membershipEvent,
+        replaceEntitiesOperation,
+        updateOperation,
+      );
+    }
+
+    return;
+  }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.supportsEventTopics} */
+  supportsEventTopics(): string[] {
+    return ['github.organization', 'github.team', 'github.membership'];
+  }
+
+  private canHandleEvents(): boolean {
+    if (this.options.catalogApi && this.options.tokenManager) {
+      return true;
+    }
+
+    // throw only once
+    if (!this.eventConfigErrorThrown) {
+      this.eventConfigErrorThrown = true;
+      throw new Error(
```

Comment:
Is it okay to throw here or is log.warn/error really what we want instead? I'm thinking if this causes the rest of the execution to stop.

Reply:
Same discussion from [here](https://github.com/backstage/backstage/pull/13931#discussion_r1001767291).
The best option should throw this error at backend start 🤔

Reply:
Thanks for sharing the past discussion, lets keep it 👍